### PR TITLE
Configure deprovision API's

### PR DESCRIPTION
**Pull request by MKodde**
The API's can be configured in the `parameters.yml` file. Sensible defaults have been added to the corresponding `dist` file.

* An outline for the Guzzle clients have been created
* DI Extension is responsible for configuring the clients and their collection.

### CODE REVIEW

File: src/OpenConext/UserLifecycle/Domain/Client/DeprovisionClientInterface.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     * @param bool $dryRun
+     */
+    public function deprovision(CollabPersonId $user, $dryRun = false);
+
+    /**
+     * Can be used to gather information about a user on the OpenConext platform.
+     *
+     * @param CollabPersonId|null $user
+     */
+    public function information(CollabPersonId $user);
```
Comment (mroest) [inline src/OpenConext/UserLifecycle/Domain/Client/DeprovisionClientInterface.php:38]: Does this getter? really belongs to this interface?  My initial feelings are this belongs to a more generic user client. What does the method do? Is it a getter?

File: src/OpenConext/UserLifecycle/Domain/ValueObject/CollabPersonId.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\UserLifecycle\Domain\ValueObject;
+
+use Webmozart\Assert\Assert;
```
Comment (mroest) [inline src/OpenConext/UserLifecycle/Domain/ValueObject/CollabPersonId.php:21]: Curious how you're going to resolve this when using deptrac :)

File: src/OpenConext/UserLifecycle/Infrastructure/UserLifecycleBundle/Client/DeprovisionClientCollection.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        foreach ($this->clients as $client){
+            $client->deprovision($user, $dryRun);
+        }
+    }
+
+    public function information(CollabPersonId $user)
+    {
+        foreach ($this->clients as $client){
+            $client->information($user);
```
Comment (mroest) [inline src/OpenConext/UserLifecycle/Infrastructure/UserLifecycleBundle/Client/DeprovisionClientCollection.php:42]: How should this work? Does it return something? Assuming it does not enrich the value object.

**Review by mroest — COMMENTED** [review]
PR looks good. You could argue to add tests for the deprovisionclientcollection and integration/acceptance test to see whether tagged services from the config end up in the collectionservice.

**Comment by MKodde** [comment]
Thanks for the review! See the reactions in the comments. I'll look more closely into testing the DI magic as this is some more error-prone code!

File: src/OpenConext/UserLifecycle/Domain/Client/DeprovisionClientInterface.php (hunk 1 above)
Comment (MKodde) [reply src/OpenConext/UserLifecycle/Domain/Client/DeprovisionClientInterface.php:38]: `information` is one of the endpoints on the service deprovision api's. It made sense to me to include it in the DeprovisionClientInterface. See it as the output of a deprovision dry run (what will be removed of the user)

File: src/OpenConext/UserLifecycle/Domain/ValueObject/CollabPersonId.php (hunk 2 above)
Comment (MKodde) [reply src/OpenConext/UserLifecycle/Domain/ValueObject/CollabPersonId.php:21]: Yeah, deptrack will not like this at all :) Will fix this when the time comes

File: src/OpenConext/UserLifecycle/Infrastructure/UserLifecycleBundle/Client/DeprovisionClientCollection.php (hunk 3 above)
Comment (MKodde) [reply src/OpenConext/UserLifecycle/Infrastructure/UserLifecycleBundle/Client/DeprovisionClientCollection.php:42]: This was just roughed in while writing the code. Some decoration will be done at this point to create an aggregation of the received json documents.

**Pull request merged**